script/wrapped-message-tests.js

```diff
@@ -0,0 +1,159 @@
+ */
+function cleanup(instance, done, fn) {
+    instance.connection.query("DELETE FROM ??;",
+        [
+            instance.config.table
+        ], (cleanupErr, data) => {
+            if (cleanupErr) {
+                console.error("There was a problem wiping the database.");
+                console.error("Other tests may be affected by this failure.");
+                done(cleanupErr);
+            }
+
+            instance.removeAllListeners();
+            console.log("Cleanup was successful");
+
+            // This should be 0 in the case of remove, which is why the assert
+            // isn't run for remove.
+            if (fn && (fn === "heartbeat" || fn === "requeue")) {
+                assert(data.affectedRows);
+            }
+
+            runNext(doneCb);
+        }
+    );
+}
+
+
+/**
+ * Inserts a record into the database using the provided query. On error, the
+ * database is wiped and the error is handled. If successful, the polling is
+ * initiated.
+ *
+ * @param {Object} instance
+ * @param {string} query
+ * @param {Function} done
+ */
+function prepTest(instance, query, done) {
+    instance.connection.query(query,
+        [
+            instance.config.table,
+            instance.owner
+        ], (prepErr, prepData) => {
+            if (prepErr) {
+                done(prepErr);
+                cleanup(instance, done);
+
+                return;
+            }
+
+            console.log("CheckNow prep was successful", prepData);
+            instance.startListening();
+        }
+    );
+}
+
+
+/**
+ * Instantiates and preps the plugin for testing. The "data" event listener will
+ * call the wrapped message function and initiate cleanup on success.
+ *
+ * @param {Function} fn The function that is to be tested.
+ * @param {string} query
+ * @param {Function} done
+ */
+function wrappedMessageTest(fn, query, done) {
+    var instance;
+
+    instance = new Plugin(config);
+    instance.on("data", (data) => {
+        console.log("CheckNow data listener activated.");
+        instance.stopListening();
+        data[fn]((err, fnData) => {
+            if (err) {
+                done(err);
+            }
+
+            console.log(`${fn} test was successful.`);
+            console.log(`${fn} data:`);
+
+            // This should be undefined for remove.
+            console.log(fnData);
+            cleanup(instance, done, fn);
+        });
+    });
+    instance.on("error", (err) => {
+        done(err);
+        cleanup(instance, done);
+    });
+    instance.connect((err) => {
+        if (err) {
+            done(err);
+
+            return;
+        }
+
+        prepTest(instance, query, done);
+    });
+}
+
+runNext = function (done) {
+    testCounter += 1;
+
+    if (tests[testCounter]) {
+        wrappedMessageTest(tests[testCounter].name, tests[testCounter].query, done);
+    } else {
+        assert.equal(errCount, 0, `Error Count: ${errCount}`);
+        process.exit(0);
```

Comment:
Indeed just copied the tests over. gitignore is there but not the linter files. Added them. I will add a directive for it. The way it was written it would just keep running runNext over and over again and would hang so I added that line instead of rewriting the test.